File: elixir/rna-transcription/lib/rna_transcription.ex
defmodule RnaTranscription do
  @doc """
  Transcribes a character list representing DNA nucleotides to RNA

  ## Examples

    iex> RnaTranscription.to_rna(~c"ACTG")
    ~c"UGAC"
  """
  @map %{?A => ?U, ?C => ?G, ?G => ?C, ?T => ?A}

  @spec to_rna([char]) :: [char]
  def to_rna(dna) do
    dna |> Enum.map(&@map[&1])
  end
end

File: go/rna-transcription/rna_transcription.go
package strand

var rnaMap = map[rune]rune{
	'G': 'C',
	'C': 'G',
	'T': 'A',
	'A': 'U',
}

func ToRNA(dna string) string {
	rna := ""
	for _, r := range dna {
		rna += string(rnaMap[r])
	}

	return rna
}
